### PR TITLE
Added Invoke-AsyncFileDownload function

### PR DESCRIPTION
**Pull request by djust270**
I was interested in your script for using via command line. I noticed you were downloading the ISO with Invoke-Webrequest which is very slow. I added the function "Invoke-AsyncFileDownload" which leverages the System.Web.Client DownloadFileTaskAsync method which is much faster and more robust for downloading files https://learn.microsoft.com/en-us/dotnet/api/system.net.webclient.downloadfiletaskasync?view=netframework-4.7.1

### CODE REVIEW

**Comment by Hrxn** [comment]
Yeah.. no. That's a myth.
iwr is not really slower, especially not if you set `$ProgressPreference = 'SilentlyContinue'` for silent output.
Don't take my word, benchmark it four yourself.

I fail to see how Async would help here, to be honest.
Besides, if we're talking about leveraging newer features of the dotnet platform, why not go all the way and use System.Net.Http.HttpClient over System.Net.WebClient?

**Comment by djust270** [comment]
Well I stand corrected then. I do like the progress bar I wrote into that function. Im not familiar with Http.HttpClient, Ill look it up in the docs. Would there a benefit to using HttpClient over WebClient? I chose the AsyncFileDownloadTask since it downloads asynchronously without needing to use Start-Job.

**Comment by pbatard** [comment]
Okay, as the main author of Fido, let me weigh in on this:

While I appreciate the effort, I am reluctant to add extra code for the commandline mode, because, since my main purpose for Fido is its use with Rufus, I see commandline mode more as a liability than anything else, and therefore, I want to keep it as simple as possible for maintainability reasons (read: I don't want to be forced to spend and extended amount of time rewriting commandline code if, say, Microsoft happens to break it with a new version of PowerShell... which, in the Microsoft PowerShell world is unfortunately not a far fetched probability).

That is actually the main reason why I went for the straightforward, simple, and unlikely to get broken `Invoke-WebRequest`, with the thinking that, if some people think that this is not enough, they can always use the `-GetUrl` option of commandline Fido, and pipe that to whatever means of downloading a file they like better.

In short I consider that `Invoke-WebRequest` is "good enough` for our purpose here, and, as such, I am reluctant to replace it with something else. Oh, and I'll point out that, in its current state, Fido is designed to be compatible with Windows 7 as well as PowerShell 7.x, whereas (since I have not tested it) I have no idea whether this PR might break things for these systems... and, as you can imagine, I am not that eager to spend time finding that out.

As such, while I am very thankful for the patch, and it does sadden me to shut down a PR when I am well aware of the effort that goes into it, I'm afraid I will have to reject this one for now... Sorry!

**Comment by djust270** [comment]
@pbatard No worries, I understand where you are coming from and thank you for making some amazing tools!
